packages/scandipwa/src/component/ProductCompareItem/ProductCompareItem.container.js

```diff
@@ -121,11 +121,10 @@ export class ProductCompareItemContainer extends PureComponent {
     }
 
     getLinkTo() {
-        const { product: { url }, product } = this.props;
+        const { product: { url } } = this.props;
 
         return {
-            pathname: url,
-            state: { product }
+            pathname: url
```

Comment:
We should pass state to history api everywhere it's possible.
This gives an opportunity to render some product page data before we have requested product from the server.